engine/access/rpc/backend/backend_transactions.go

```diff
@@ -737,6 +731,14 @@ func (b *backendTransactions) getTransactionResultsByBlockIDFromAnyExeNode(
 		}
 		errs = multierror.Append(errs, err)
 	}
+
+	// if we were passed 0 execution nodes add a specific error
+	if len(execNodes) == 0 {
```

Comment:
this should go above the loop and you can simplify by must returning the error directly instead of adding it to `errs`